initrd/bin/kexec-unseal-key

```diff
@@ -16,27 +16,55 @@ key_file="$1"
 if [ -z "$key_file" ]; then
 	key_file="/tmp/secret/secret.key"
 fi
+if [ "$CONFIG_TPM" = "y" ];then
+	tpm nv_readvalue \
+	    -in "$TPM_INDEX" \
+	    -sz "$TPM_SIZE" \
+	    -of "$sealed_file" \
+		|| die "Unable to read key from TPM NVRAM"
 
-tpm nv_readvalue \
-	-in "$TPM_INDEX" \
-	-sz "$TPM_SIZE" \
-	-of "$sealed_file" \
-|| die "Unable to read key from TPM NVRAM"
+	for tries in 1 2 3; do
+		if [ "$CONFIG_AUTO_UNLOCK" != y ]; then
+			read -s -p "Enter unlock password (blank to abort): " tpm_password
+			echo
+			if [ -z "$tpm_password" ]; then
+				die "Aborting unseal disk encryption key"
+			fi
 
-for tries in 1 2 3; do
-	read -s -p "Enter unlock password (blank to abort): " tpm_password
-	echo
+			tpm unsealfile \
+			    -if "$sealed_file" \
+			    -of "$key_file" \
+			    -pwdd "$tpm_password" \
+			    -hk 40000000
+		else
+			tpm unsealfile \
+			    -if "$sealed_file" \
+			    -of "$key_file" \
+			    -hk 40000000
+		fi
 
-	if [ -z "$tpm_password" ]; then
-		die "Aborting unseal disk encryption key"
+		if [ $? -eq 0 ]; then
+			# should be okay if this fails
+			shred -n 10 -z -u /tmp/secret/sealed 2> /dev/null || true
+			exit 0
+		fi
+
+		pcrs
+		warn "Unable to unseal disk encryption key"
+	done
+elif [ "$CONFIG_TPM2_TOOLS" = "y" ]; then
+	if [ "$CONFIG_ATTEST_TOOLS" = "y" ]; then
+		echo "Bring up network for remote attestation"
+		network-init-recovery
+	fi
+	for tries in 1 2 3; do
+	if [ "$CONFIG_AUTO_UNLOCK" = "y" ]; then
+		tpmr unseal "0x8100000$TPM_INDEX" "sha256:0,1,2,3,4,5,6,7" > "$key_file"
+	else
+		tpmr unseal "0x8100000$TPM_INDEX" "sha256:0,1,2,3,4,5,6,7" "file:-" > "$key_file"
```

Comment:
Doesn't unseal as sealed. Either PCRs 0-7 or file here, but both seems to be problematic. 
Insights?